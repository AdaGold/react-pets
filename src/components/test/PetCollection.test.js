import React from 'react';
import PetCollection from '../PetCollection';
import { shallow, mount } from 'enzyme';

const PETS = [
  {
    name: "test pet 1",
    breed: "breed"
  },
  {
    name: "test pet 2",
    breed: "breed"
  }
];

describe('PetCollection', () => {
  test('that it matches an existing snapshot', () => {
    // First Mount the Component in the testing DOM
    // Arrange

    const pets = PETS;
    const wrapper = shallow(
      <PetCollection
        pets={pets}
        />
    );

    // Assert that it looks like the last snapshot
    expect(wrapper).toMatchSnapshot();
  });

  test('renders without crashing even with no pets', () => {
    // First Mount the Component in the testing DOM
    // Arrange

    const pets = [];
    const wrapper = shallow(
      <PetCollection
        pets={pets}
        />
    );

    // Assert that it looks like the last snapshot
    expect(wrapper).toMatchSnapshot();
  });

  test('successfully deep mounts', () => {
    // First Mount the Component in the testing DOM
    // Arrange
    const wrapper = mount(
      <PetCollection
        pets={PETS}
        />
    );

    // Assert that it looks like the last snapshot
    expect(wrapper).toMatchSnapshot();

    wrapper.unmount();
  });
});
