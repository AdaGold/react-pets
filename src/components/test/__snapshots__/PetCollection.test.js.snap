// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PetCollection renders without crashing even with no pets 1`] = `
<section
  className="pets"
>
  <table>
    <tbody>
      <tr>
        <th>
          Name
        </th>
        <th>
          Breed
        </th>
        <th>
          Age
        </th>
        <th>
          About
        </th>
      </tr>
    </tbody>
  </table>
  <NewPetForm
    addPetCallback={[Function]}
  />
</section>
`;

exports[`PetCollection successfully deep mounts 1`] = `
<PetCollection
  pets={
    Array [
      Object {
        "breed": "breed",
        "name": "test pet 1",
      },
      Object {
        "breed": "breed",
        "name": "test pet 2",
      },
    ]
  }
>
  <section
    className="pets"
  >
    <table>
      <tbody>
        <tr>
          <th>
            Name
          </th>
          <th>
            Breed
          </th>
          <th>
            Age
          </th>
          <th>
            About
          </th>
        </tr>
        <Pet
          breed="breed"
          key="0"
          name="test pet 1"
        >
          <tr
            className="pet"
          >
            <td>
              test pet 1
            </td>
            <td>
              breed
            </td>
            <td />
            <td />
          </tr>
        </Pet>
        <Pet
          breed="breed"
          key="1"
          name="test pet 2"
        >
          <tr
            className="pet"
          >
            <td>
              test pet 2
            </td>
            <td>
              breed
            </td>
            <td />
            <td />
          </tr>
        </Pet>
      </tbody>
    </table>
    <NewPetForm
      addPetCallback={[Function]}
    >
      <form
        className="dan-class"
        id="new-pet"
        onSubmit={[Function]}
      >
        <div>
          <label
            htmlFor="name"
          >
            Pet Name
          </label>
          <input
            name="name"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
        <div>
          <label
            htmlFor="breed"
          >
            Breed
          </label>
          <input
            name="breed"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
        <div>
          <label
            htmlFor="age"
          >
            Age
          </label>
          <input
            name="age"
            onChange={[Function]}
            pattern="[0-9]*"
            type="number"
            value=""
          />
        </div>
        <div>
          <label
            htmlFor="about"
          >
            About
          </label>
          <input
            name="about"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
        <input
          type="submit"
          value="Add Pet"
        />
      </form>
    </NewPetForm>
  </section>
</PetCollection>
`;

exports[`PetCollection that it matches an existing snapshot 1`] = `
<section
  className="pets"
>
  <table>
    <tbody>
      <tr>
        <th>
          Name
        </th>
        <th>
          Breed
        </th>
        <th>
          Age
        </th>
        <th>
          About
        </th>
      </tr>
      <Pet
        breed="breed"
        key="0"
        name="test pet 1"
      />
      <Pet
        breed="breed"
        key="1"
        name="test pet 2"
      />
    </tbody>
  </table>
  <NewPetForm
    addPetCallback={[Function]}
  />
</section>
`;
